pkg/plugins/storage/storage/storage.go

```diff
@@ -179,6 +198,85 @@ func (s *StorageStorageService) Delete(bucket string, key string) error {
 	return nil
 }
 
+func (s *StorageStorageService) PreSignUrl(bucket string, key string, operation plugin.Operation, expiry uint32) (string, error) {
+	newErr := errors.ErrorsWithScope(
+		"StorageStorageService.PreSignUrl",
+		map[string]interface{}{
+			"bucket":    bucket,
+			"key":       key,
+			"operation": operation.String(),
+		},
+	)
+	if len(bucket) == 0 {
+		return "", newErr(
+			codes.InvalidArgument,
+			"provide non-blank bucket",
+			fmt.Errorf(""),
+		)
+	}
+	if len(key) == 0 {
+		return "", newErr(
+			codes.InvalidArgument,
+			"provide non-blank key",
+			fmt.Errorf(""),
+		)
+	}
+	if expiry > 604800 {
+		return "", newErr(
+			codes.InvalidArgument,
+			"expiry time can't be longer than 604800 seconds (7 days)",
+			fmt.Errorf("expiry provided: %d", expiry),
+		)
+	}
+	serviceAccount := utils.GetEnv("SERVICE_ACCOUNT_LOCATION", "")
+	if len(serviceAccount) == 0 {
+		return "", newErr(
+			codes.InvalidArgument,
+			"provide non-blank service account config path",
+			fmt.Errorf("path provided: %s", serviceAccount),
+		)
+	}
+	jsonKey, err := ioutil.ReadFile(serviceAccount)
+	if err != nil {
+		return "", newErr(
+			codes.Internal,
+			"error reading service account config file at given SERVICE_ACCOUNT_LOCATION",
+			err,
+		)
+	}
+	conf, err := s.utils.JWTConfigFromJSON(jsonKey)
+	if err != nil {
+		return "", newErr(
+			codes.Internal,
+			"error getting JWT from service account JSON",
+			err,
+		)
+	}
+	var headers []string
+	if operation == plugin.WRITE { // If its a put operation, need to add additional headers
+		headers = append(headers, "Content-Type:application/octet-stream")
+	} else {
+		operation = plugin.READ
+	}
+	opts := &storage.SignedURLOptions{
+		Scheme:         storage.SigningSchemeV4,
+		Method:         []string{"GET", "PUT"}[operation],
```

Comment:
I feel like translation between operations `READ` and `WRITE` to `"GET"` or `"PUT"` should be more explicitly defined instead of relying on the array orders matching.